### DISCUSSION TITLE
Getting pandoc command line arguments from a filter or a custom writer

### DISCUSSION THREAD

**Discussion by massifrg** [jgm/pandoc#9689] [Q&A]
I'd like to get the arguments with which `pandoc` has been called from within a filter.

In particular I'd like to know the value of the `-f` (`--from`) option, to know the format of the input source.

Something like the FORMAT global variable -- that is set to the desired output format (or custom writer) --, but relative to the input format.

**Comment by jgm** [comment]
I'm not sure the input source format is made available anywhere!  @tarleb is that right, and perhaps it should be?

However, the source file names are made available in a variable, `sourcefile`.  (Variables are accessible through WRITER_OPTIONS, so something like `WRITER_OPTIONS.variables.sourcefile` should work.)  This could be null (stdin), a string (one file), or a table (multiple files).

**Reply by massifrg** [reply]
I'm working on my [pandoc-include-doc](https://github.com/massifrg/pandoc-include-doc) project, that has a custom writer -- [inclusion-tree.lua](https://github.com/massifrg/pandoc-include-doc#extracting-the-structure-of-inclusions-with-inclusion-treelua) -- that produces a JSON representation of the tree structure of a document that includes other documents.

Every node in that tree has these properties:

```json
    "format": "html",
    "id": "root",
    "sha1": "64b297779956f64503df8dccca191f76403462f0",
    "src": "master.html",
    "children": [
      ...
    ]
```

The goal of my request is to set the `format` of the root node.

Of course it could be inferred from the `src` property, but reading it from the `--from` option would work even when the source file has no extension or the the source is the standard input.

Moreover, the extension and the format don't always coincide.

**Reply by tarleb** [reply]
> I'm not sure the input source format is made available anywhere! @tarleb is that right, and perhaps it should be?

Right, it's currently not exposed to Lua. And I agree that it might be useful for filters.

The separation of readers and writers would be weakened somewhat if this kind of information was made available to custom readers or writers. I tend to think that such information should be passed explicitly, and by the user. The "format extension" system is well suited for that.
